pkg/utils/userip_test.go

```diff
@@ -0,0 +1,31 @@
+package utils
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/grpc/peer"
+
+	"github.com/openservicemesh/osm/pkg/tests"
+)
+
+func TestGetIPFromContext(t *testing.T) {
+	assert := assert.New(t)
+
+	ctxs := []context.Context{
+		context.Background(),
+		peer.NewContext(context.TODO(), &peer.Peer{Addr: tests.NewMockAddress("9.8.7.6")}),
+	}
+
+	for _, ctx := range ctxs {
+		result := GetIPFromContext(ctx)
+
+		if result == nil {
+			assert.Nil(result)
```

Comment:
I don't think you need this condition and check. I think instead you'd probably just want to assert that result is not nil